pkg/secretless/plugin/sharedobj/available_plugins_test.go

```diff
@@ -16,26 +19,111 @@ type mockTCPPlugin struct{ tcp.Plugin }
 // These need to sit outside of the func since we are comparing them in the
 // tests
 var mockHTTPPlugins = map[string]http.Plugin{
-	"one": mockHTTPPlugin{},
-	"two": mockHTTPPlugin{},
+	"one": &mockHTTPPlugin{},
+	"two": &mockHTTPPlugin{},
 }
 
 var mockTCPPlugins = map[string]tcp.Plugin{
-	"one":   mockTCPPlugin{},
-	"two":   mockTCPPlugin{},
-	"three": mockTCPPlugin{},
+	"one":   &mockTCPPlugin{},
+	"two":   &mockTCPPlugin{},
+	"three": &mockTCPPlugin{},
 }
 
-func getMockPlugins() plugin.AvailablePlugins {
+func mockPlugins() plugin.AvailablePlugins {
 	return &Plugins{
 		HTTPPluginsByID: mockHTTPPlugins,
 		TCPPluginsByID:  mockTCPPlugins,
 	}
 }
 
+func mockInternalPlugins() plugin.AvailablePlugins {
```

Comment:
Naming nit: 

Probably even better is to create a mock subpackage, which will clean up this test file and let you rely on the package modifier for naming.... mock.InternalPlugins() etc.

Kumbi's been using this pattern in some places.